### DISCUSSION TITLE
Sign SegWit transaction without downloading blocks

### DISCUSSION THREAD

**Discussion by liorwavebl** [btcsuite/btcd#2110] [Q&A]
I managed to sign a SegWit transaction. I have two servers- one with access to the blockchain and the second has no access at all. The second server can get requests from the first server.
Can I sign a transaction from the second server without downloading blocks?

**Comment by guggero** [comment]
You can use PSBTs (https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki) for this. On the node with blockchain access you would add all required information (mot notably the witness UTXO field `PSBT_IN_WITNESS_UTXO`) for each input. The signing node then just uses that meta information to sign.
Although `btcd` itself doesn't have PSBT support yet, so you'll need to write code yourself or use `lnd` as the on-chain wallet.

**Reply by liorwavebl** [reply]
Can I send the data from the one who has access to the blockchain to the second one without use PSBT?
If I used non-segwit I can access the UTXO `scriptPubKey` and sign the transaction with it by sending it from the first server to the second one.
How can I do it with SegWit transactions?

**Reply by guggero** [reply]
Sure, you can use whatever method you want to send the UTXO information from one server to another. Using a PSBT is just the standard way to do it.

**Reply by liorwavebl** [reply]
How can I extract the equivalent data to `scriptPubKey` for SegWit transactions?
ASFAIU it's called `witnessScript`. I cannot get the it by calling `decoderawtransaction`.

**Reply by guggero** [reply]
Should be `gettxout`.

**Reply by liorwavebl** [reply]
According to bitcoin API `gettxout` returns the `scriptPubKey` from the chain. If i passed the `scriptPubKey` to  `txscript.WitnessSignature` I'm getting an exception that says: `non-mandatory-script-verify-flag (Witness requires empty scriptSig)`
